core/src/test/java/com/linecorp/armeria/server/ServerChildChannelPipelineCustomizerTest.java

```diff
@@ -0,0 +1,120 @@
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.ResponseTimeoutException;
+import com.linecorp.armeria.client.SessionProtocolNegotiationException;
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.common.util.SslContextUtil;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.traffic.ChannelTrafficShapingHandler;
+
+class ServerChildChannelPipelineCustomizerTest {
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.childChannelPipelineCustomizer(
+                    pipeline ->  pipeline.addFirst(new ChannelTrafficShapingHandler(1024, 0)));
+            sb.service("/", new AbstractHttpService() {
+                @Override
+                protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req)
+                        throws Exception {
+                    return HttpResponse.of(
+                        ResponseHeaders.of(HttpStatus.OK, "header1", Strings.repeat("a", 2048)));
+                }
+            });
+        }
+    };
+
+    static final SSLEngine sslEngine =  SslContextUtil.createSslContext(
+            SslContextBuilder::forClient,
+            /* forceHttp1 */ true,
+            /* tlsAllowUnsafeCiphers */ false,
+            ImmutableList.of()).newEngine(ByteBufAllocator.DEFAULT);
+
+    @RegisterExtension
+    static ServerExtension server2 = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.http(8080);
+            sb.childChannelPipelineCustomizer(
+                    pipeline ->  pipeline.addFirst(new SslHandler(sslEngine)));
+            sb.service("/", (ctx, req) -> HttpResponse.of("content"));
+        }
+    };
+
+    @EnumSource(value = SessionProtocol.class, names = {"H1", "H1C"})
+    @ParameterizedTest
+    void testResponseTimeout(SessionProtocol protocol) {
```

Comment:
Because we added `SslHandler` to the pipeline, `server2` should be an HTTPS server now. Therefore, you only need to test `H1`. `H1C` is not very relevant.